src/memgraph.cpp

```diff
@@ -522,13 +522,19 @@ class BoltSession final : public memgraph::communication::bolt::Session<memgraph
       : memgraph::communication::bolt::Session<memgraph::communication::v2::InputStream,
                                                memgraph::communication::v2::OutputStream>(input_stream, output_stream),
         db_(data->db),
+        interpreter_context_(data->interpreter_context),
         interpreter_(data->interpreter_context),
         auth_(data->auth),
 #if MG_ENTERPRISE
         audit_log_(data->audit_log),
 #endif
         endpoint_(endpoint),
         run_id_(data->run_id) {
+    interpreter_context_->interpreters.WithLock([this](auto &interpreters) { interpreters.insert(&interpreter_); });
```

Comment:
The `insert` and `erase` might produce some issues on the `mgbench` workload, so we should definitely run `mgbench` and observe if there are some degradations 😄